pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -577,11 +577,101 @@ func (r *Reconciler) processTargetPools(desired bool, poolFunc poolProcessor) er
 	return nil
 }
 
+func (r *Reconciler) ensureInstanceGroup(instanceGroupName string) error {
+	// Get an instance group so we can check that it does in fact exist
+	_, err := r.computeService.InstanceGroupGet(r.projectID, r.providerSpec.Zone, instanceGroupName)
+	if isNotFoundError(err) {
+		// Handle the creation of a new instance group
+		if err := r.registerNewInstanceGroup(); err != nil {
+			return fmt.Errorf("failed to register the new instance group named %s: %v", instanceGroupName, err)
+		}
+	} else if err != nil {
+		return err
+	}
+
+	registered, err := r.checkRegistrationOfBackend()
+	if err != nil {
+		return err
```

Comment:
Make sure to wrap errors with additional context

Reply:
Done